ls8/cpu.py

```diff
@@ -83,51 +75,66 @@ def trace(self):
         """
 
         print(f"TRACE: %02X | %02X %02X %02X |" % (
-            self.pc,
+            self.PC,
             #self.fl,
             #self.ie,
-            self.ram_read(self.pc),
-            self.ram_read(self.pc + 1),
-            self.ram_read(self.pc + 2)
+            self.ram_read(self.PC),
+            self.ram_read(self.PC + 1),
+            self.ram_read(self.PC + 2)
         ), end='')
 
         for i in range(8):
             print(" %02X" % self.reg[i], end='')
 
         print()
 
+    def handle_HLT(self):
+        self.PC = 0
+        return 'HLT'
+    
+    def handle_LDI(self):
+        operand_a = self.PC + 1
+        operand_b = self.PC + 2
+        address = self.ram_read(operand_a)
+        value = self.ram_read(operand_b)
+        self.reg[address] = value
+        self.PC += 3
+
+    def handle_PRN(self):
+        address = self.ram_read(self.PC + 1)
+        print(self.reg[address])
+        self.PC +=2
+
+    def handle_MUL(self):
```

Comment:
Where can we put this to compose the code better? Hint: there's a special place for arithmetic operations...